src/Odelstm.jl

```diff
@@ -0,0 +1,496 @@
+using Lux
+using ComponentArrays
+using Optimisers
+using LinearAlgebra
+using Random
+using Statistics
+using MLDatasets
+using DelimitedFiles
+using Printf
+
+export ODELSTMCell, ODELSTMModel, IrregularSequenceLearner
+export PersonData, ETSMnistData, XORData
+export train_model!, evaluate_model, load_dataset
+
+mutable struct ODELSTMCell{F,S}
+    lstm_cell::Lux.LSTMCell
+    f_node::F
+    solver::S
+    input_size::Int
+    hidden_size::Int
+end
+
+function ODELSTMCell(input_size::Int, hidden_size::Int, solver)
+    lstm_cell = Lux.LSTMCell(input_size, hidden_size)
+    f_node = Chain(
+        Dense(hidden_size, hidden_size, tanh),
+        Dense(hidden_size, hidden_size)
+    )
+    return ODELSTMCell(lstm_cell, f_node, solver, input_size, hidden_size)
+end
+
+function (cell::ODELSTMCell)(x, state, ts, p, st)
+    h, c = state
+    new_h, new_c, st = cell.lstm_cell(x, (h, c), p, st)
+    
+    if !(cell.solver isa Union{Tsit5,DP5,BS3})
+        evolved_h, st = solve_fixed_step(cell, new_h, ts, p, st)
+    else
+        evolved_h, st = solve_adaptive(cell, new_h, ts, p, st)
+    end
+    
+    return evolved_h, (evolved_h, new_c), st
+end
+
+function solve_fixed_step(cell::ODELSTMCell, h, ts, p, st)
+    prob = ODEProblem((u,p,t)->cell.f_node(u,p,st)[1], h, (0.0f0, Float32(ts)))
+    sol = solve(prob, cell.solver; adaptive=false)
```

Comment:
No need to make completely separate, just allow a keyword argument in `ODELSTMCell` and just use the bool in the other function, delete the extra code.